core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleWriterSuite.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.sort
+
+import java.io.File
+
+import org.mockito.Mockito._
+import org.mockito.MockitoAnnotations
+import org.scalatest.BeforeAndAfterEach
+import org.apache.spark.{ShuffleDependency, _}
+import org.apache.spark.memory.MemoryTestingUtils
+import org.apache.spark.serializer.JavaSerializer
+import org.apache.spark.shuffle.{BaseShuffleHandle, IndexShuffleBlockResolver}
+import org.apache.spark.util.Utils
+
+class SortShuffleWriterSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private val shuffleId = 0
+  private val numMaps = 5
+  private var outputFile: File = _
+  private val conf: SparkConf = new SparkConf(loadDefaults = false)
+  private val sc = new SparkContext("local-cluster[1,1,1024]", "test", conf)
+  private var shuffleHandle: BaseShuffleHandle[Int, Int, Int] = _
+  private val shuffleBlockResolver = new IndexShuffleBlockResolver(conf)
+  private val serializer = new JavaSerializer(conf)
+  private val context = MemoryTestingUtils.fakeTaskContext(sc.env)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    MockitoAnnotations.initMocks(this)
+    val partitioner = new Partitioner() {
+      def numPartitions = numMaps
+      def getPartition(key: Any) = Utils.nonNegativeMod(key.hashCode, numPartitions)
+    }
+    shuffleHandle = {
+      val dependency = mock(classOf[ShuffleDependency[Int, Int, Int]])
+      when(dependency.partitioner).thenReturn(partitioner)
+      when(dependency.serializer).thenReturn(serializer)
+      when(dependency.aggregator).thenReturn(None)
+      when(dependency.keyOrdering).thenReturn(None)
+      new BaseShuffleHandle(shuffleId, numMaps = numMaps, dependency)
+    }
+  }
+
+  override def afterEach(): Unit = {
+    try {
+      if (outputFile != null) {
+        Utils.deleteRecursively(outputFile)
+      }
+    } finally {
+      super.afterEach()
+    }
+  }
+
+  test("write empty iterator") {
+    val writer = new SortShuffleWriter[Int, Int, Int](
+      shuffleBlockResolver,
+      shuffleHandle,
+      mapId = 1, // MapId
```

Comment:
delete the comment `// MapId` as you are already using named arguments it does not add additional value